src/test_suite/test_suite.py

```diff
@@ -485,5 +488,30 @@ def decode_protobuf(
     print(f"{sum(write_results)} files successfully written")
 
 
+def set_ld_preload_asan():
+    # Run ldconfig -p and capture output
+    import subprocess
+
+    ldconfig_output = subprocess.check_output(["ldconfig", "-p"], text=True)
+
+    # Filter lines containing "asan"
+    asan_libs = [line for line in ldconfig_output.split("\n") if "asan" in line]
+
+    # Extract the first library path if available
+    if asan_libs:
+        first_asan_lib = asan_libs[0].split()[-1]
+    else:
+        print("No ASAN library found.")
+        return
+
+    # Set LD_PRELOAD environment variable
+    import os
+    import sys
+
+    os.environ["LD_PRELOAD"] = first_asan_lib
+    print(f"LD_PRELOAD set to {first_asan_lib}")
+    os.execvp(sys.argv[0], sys.argv)
```

Comment:
Can you confirm that this doesn't terminate the main process?

Reply:
it restarts the process - I confirmed it works